### ISSUE TITLE
The path to code coverage for transactions

### ISSUE THREAD

**Issue by jmeas** [jmeas/moolah#473]
Covering the transactions section of the app will be a big win. I wouldn't be surprised if that alone would take us to 85%+ coverage.

There are a few things that need to be done first, though.

- The components depend on the old Redux store, which needs to be refactored to use the new system
- If I use the new system, I might as well make it XHR based rather than setTimeout based
- If I refactor it that much, I might as well update the mock endpoints to receive a date, rather than doing "client-side" filtering

So there's quite a bit to do before I can write tests for the components. In order, the steps are:

- [x] plan the new transactions API
- [x] create mock server for that mock API
- [x] update redux to interact with mock API
- [x] write tests for the new redux stuff
- [x] update components
- [x] write tests for components
- [ ] update API to be more powerful

Let's do it ✊ 

---

**The mock transactions API*

```
GET /transactions/:transactionId
```

Query params:

```
// the page to display
page[number]=1

// the number of results per page
page[size]=25

// sort by descending date and then by description
sort=-date,description
```

**Comment by jmeas** [comment]
I was able to copy + paste the work done in Categories over to Transactions. This should be a good place to start from.

At first, I thought I'd *need* to require server side filtering to build the UI, but since I plan to cache the transactions in the store, I think I can get around this.

**Comment by jmeas** [comment]
gonna close this. The transactions contents page has been tested. The rest is dependent on other work that's been merged into #241
